src/Item_Rack.php

```diff
@@ -589,28 +589,24 @@ public function showForm($ID, array $options = [])
        //get all used items
         $used = $used_reserved = [];
         $iterator = $DB->request([
-            'FROM' => $this->getTable()
+            'SELECT' => ['itemtype', 'items_id', 'is_reserved'],
+            'FROM' => static::getTable()
         ]);
         foreach ($iterator as $row) {
-            $used[$row['itemtype']][] = $row['items_id'];
+            if ($row['is_reserved']) {
+                $used_reserved[$row['itemtype']][] = $row['items_id'];
+            } else {
+                $used[$row['itemtype']][] = $row['items_id'];
```

Comment:
In previous version of the code, all items were present in `$used`, whatever their reserved state was.